juju/juju
# Add a new k8s environ provider

## Description of change

k8s gains an environ provider implementation. This is necessary to allow various credential related CLI commands to function, as well as paving the way for model deletion and pluggable CAAS providers like we do for clouds.

A consequence is that the caas agent facade no longer needs a bespoke Model() API and is reduced to being built from existing facade api plugins, allowing a bunch of code to be deleted.

## QA steps

bootstrap a cloud
deploy k8s
deploy a workload on top of k8s
ensure the various credential CLI works
ensure kill-controller etc work without errors
